src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -145,16 +161,23 @@ object SnowflakeTelemetry {
       curLogs = logs
       logs = Nil
     }
-    curLogs.foreach {
-      case (log, timestamp) =>
-        logger.debug(s"""
-             |Send Telemetry
-             |timestamp:$timestamp
-             |log:${log.toString}"
+    if (testFakeMessagesSender.isEmpty) {
+      curLogs.foreach {
+        case (log, timestamp) =>
+          logger.debug(s"""
+                          |Send Telemetry
+                          |timestamp:$timestamp
+                          |log:${log.toString}"
            """.stripMargin)
-        telemetry.asInstanceOf[TelemetryClient].addLogToBatch(log, timestamp)
+          telemetry.asInstanceOf[TelemetryClient].addLogToBatch(log, timestamp)
+      }
+      telemetry.sendBatchAsync()
+    } else {
+      // This is for test only
```

Comment:
If fake message sender is set, the telemetry message is appended into the ArrayBuffer instead of sending to snowflake, and the test can check the message to be sent.